lib/utils/cli.go

```diff
@@ -49,65 +52,91 @@ const (
 )
 
 // InitLogger configures the global logger for a given purpose / verbosity level
-func InitLogger(purpose LoggingPurpose, level logrus.Level) {
+func InitLogger(purpose LoggingPurpose, level slog.Level) {
 	logrus.StandardLogger().ReplaceHooks(make(logrus.LevelHooks))
-	logrus.SetLevel(level)
+	logrus.SetLevel(logutils.SlogLevelToLogrusLevel(level))
 	switch purpose {
 	case LoggingForCLI:
 		// If debug logging was asked for on the CLI, then write logs to stderr.
 		// Otherwise, discard all logs.
-		if level == logrus.DebugLevel {
-			debugFormatter := logutils.NewDefaultTextFormatter(trace.IsTerminal(os.Stderr))
+		if level == slog.LevelDebug {
+			enableColors := trace.IsTerminal(os.Stderr)
+			debugFormatter := logutils.NewDefaultTextFormatter(enableColors)
 			_ = debugFormatter.CheckAndSetDefaults()
 			logrus.SetFormatter(debugFormatter)
-			logrus.SetOutput(os.Stderr)
+
+			w := logutils.NewSharedWriter(os.Stderr)
+			logrus.SetOutput(w)
+			slog.SetDefault(slog.New(logutils.NewSlogTextHandler(w, &logutils.SlogTextHandlerConfig{
+				Level:        level,
+				EnableColors: enableColors,
+				WithCaller:   true,
+			})))
```

Comment:
```go
// NewDefaultTextFormatter creates a TextFormatter with
// the default options set.
func NewDefaultTextFormatter(enableColors bool) *TextFormatter {
	return &TextFormatter{
		ComponentPadding: trace.DefaultComponentPadding,
		FormatCaller:     formatCallerWithPathAndLine,
		ExtraFields:      knownFormatFieldNames,
		EnableColors:     enableColors,
		callerEnabled:    true,
		timestampEnabled: false,
	}
}
```

I notice that there are other default settings in the original `NewDefaultTextFormatter` that aren't obviously accounted for in the `NewSlogTextHandler` (e.g. `formatCallerWithPathAndLine`, `knownFormatFieldNames`) but maybe I'm just missing it). Flagging this just to double check.